[2024-11-06 15:06:30,754][root][INFO] - ---------------------------------------------------------------
[2024-11-06 15:06:30,755][root][INFO] - Env:
CUDA_VISIBLE_DEVICES: null
PYTHONOPTIMIZE: null

[2024-11-06 15:06:30,756][root][INFO] - Command line args:

[2024-11-06 15:06:30,756][root][INFO] - ---------------------------------------------------------------
[2024-11-06 15:06:30,758][root][INFO] - Run config:
device: '4'
solver_fn:
  _target_: netket.optimizer.solver.cholesky
lr: 0.001
diag_shift: 1.0e-10
n_iter: 2000
chunk_size_jac: 2
chunk_size_vmap: 1
save_every: 3
base_path: /scratch/.amisery/grad_sample/
model:
  _target_: grad_sample.models.ising.TFI
  L: 3
  h: 1.0
ansatz:
  _target_: netket.models.RBM
  alpha: 4
  param_dtype: complex
task:
  _target_: grad_sample.tasks.fullsum_analysis.FullSumPruning

[2024-11-06 15:06:30,759][root][INFO] - ---------------------------------------------------------------
[2024-11-06 15:06:33,500][jax._src.xla_bridge][INFO] - Unable to initialize backend 'rocm': module 'jaxlib.xla_extension' has no attribute 'GpuAllocatorConfig'
[2024-11-06 15:06:33,502][jax._src.xla_bridge][INFO] - Unable to initialize backend 'tpu': INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory
[2024-11-06 15:06:50,045][root][CRITICAL] - object of type 'int' has no len()
jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/main.py", line 50, in main
    task()
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 42, in __call__
    in_idx, fid_vals, dp_dist_ev, infid_ev = self.prune()
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 173, in prune
    out = prune_vmap(jnp.arange(1,self.dimH))
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 171, in <lambda>
    prune_vmap = jax.vmap(lambda i: self.prune_to(i, in_idx))
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 159, in prune_to
    dp_sol = self.compute_change(jax.lax.dynamic_slice(in_idx,0,i), vmap_version=False)
TypeError: object of type 'int' has no len()
